libnopegl/src/node_rtt.c

```diff
@@ -326,11 +331,123 @@ static int rtt_prefetch(struct ngl_node *node)
+        return 0;
+
+    struct texture *textures[NGLI_MAX_COLOR_ATTACHMENTS] = {NULL};
+    struct texture *depth_texture = NULL;
+    for (size_t i = 0; i < o->nb_color_textures; i++) {
+        textures[i] = ngli_texture_create(ctx->gpu_ctx);
+        if (!textures[i]) {
+            ret = NGL_ERROR_MEMORY;
+            goto fail;
+        }
+
+        const struct rtt_texture_info info = get_rtt_texture_info(o->color_textures[i]);
+        struct texture_params texture_params = info.texture_priv->params;
+        texture_params.width = width;
+        texture_params.height = height;
+
+        ret = ngli_texture_init(textures[i], &texture_params);
+        if (ret < 0)
+            goto fail;
+    }
+
+    if (o->depth_texture) {
+        depth_texture = ngli_texture_create(ctx->gpu_ctx);
+        if (!depth_texture) {
+            ret = NGL_ERROR_MEMORY;
+            goto fail;
+        }
+
+        const struct rtt_texture_info info = get_rtt_texture_info(o->depth_texture);
+        struct texture_params texture_params = info.texture_priv->params;
+        texture_params.width = width;
+        texture_params.height = height;
+
+        ret = ngli_texture_init(depth_texture, &texture_params);
+        if (ret < 0)
+            goto fail;
+    }
+
+    struct rtt_ctx *rtt_ctx = ngli_rtt_create(ctx);
+    if (!rtt_ctx) {
+        ret = NGL_ERROR_MEMORY;
+        goto fail;
+    }
+
+    struct rtt_params rtt_params = s->rtt_params;
+    rtt_params.width  = width;
+    rtt_params.height = height;
+
+    for (size_t i = 0; i < o->nb_color_textures; i++)
+        rtt_params.colors[i].attachment = textures[i];
+    rtt_params.depth_stencil.attachment = depth_texture;
+
+    ret = ngli_rtt_init(rtt_ctx, &rtt_params);
+    if (ret < 0)
+        goto fail;
+
+    ngli_rtt_freep(&s->rtt_ctx);
+
+    s->width = width;
+    s->height = height;
+    s->rtt_params = rtt_params;
+    s->rtt_ctx = rtt_ctx;
+
+    for (size_t i = 0; i < o->nb_color_textures; i++) {
+        const struct rtt_texture_info info = get_rtt_texture_info(o->color_textures[i]);
+        struct texture_priv *texture_priv = info.texture_priv;
+        ngli_texture_freep(&texture_priv->texture);
+        texture_priv->texture = textures[i];
+        texture_priv->image.params.width = width;
+        texture_priv->image.params.height = height;
+        texture_priv->image.planes[0] = textures[i];
+        texture_priv->image.rev = texture_priv->image_rev++;
+    }
+
+    if (o->depth_texture) {
+        const struct rtt_texture_info info = get_rtt_texture_info(o->depth_texture);
+        struct texture_priv *texture_priv = info.texture_priv;
+        ngli_texture_freep(&texture_priv->texture);
+        texture_priv->texture = depth_texture;
+        texture_priv->image.params.width = width;
+        texture_priv->image.params.height = height;
+        texture_priv->image.planes[0] = depth_texture;
+        texture_priv->image.rev = texture_priv->image_rev++;
```

Comment:
I'm assuming all the other fields (in `image` and `image.params`) are properly initialized and we don't want to reset them?

Should we assert that `image.rev` is `0` at this point? Or should we unref it first?

Reply:
> I'm assuming all the other fields (in `image` and `image.params`) are properly initialized and we don't want to reset them?

Yes.

> 
> Should we assert that `image.rev` is `0` at this point? Or should we unref it first?

`image.rev` is not a ref counter. Its purpose is to provide an unique identifier and a way to notify the texture consumer that the underlying texture has changed.